sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
      *
      * @param httpRequest The HTTP Request being sent
      * @param connection The HttpURLConnection being sent to
-     *
      * @return A HttpResponse object
      */
     private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection connection) {
         try {
             int responseCode = connection.getResponseCode();
+            Headers responseHeaders = getResponseHeaders(connection);
+            ServerSentEventListener listener = httpRequest.getServerSentEventListener();
+            RetrySSEResult retrySSEResult;
+            if (connection.getErrorStream() == null) {
+                if (isTextEventStream(responseHeaders) && listener != null) {
+                    try {
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                        retrySSEResult = processBuffer(reader, listener);
+                        if (retrySSEResult != null) {
+                            retryExceptionForSSE(retrySSEResult, listener, httpRequest);
+                        }
+                    } catch (IOException e) {
+                        throw LOGGER.logThrowableAsError(new RuntimeException(e));
+                    }
+                }
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders);
```

Comment:
This doesn't contain the response body?

Reply:
Do we expect Stream event APIs to have a body isn't that what is read into stream data?